Lib/_pydecimal.py

```diff
@@ -1244,10 +1232,9 @@ def __sub__(self, other, context=None):
         if other is NotImplemented:
             return other
 
-        if self._is_special or other._is_special:
-            ans = self._check_nans(other, context=context)
-            if ans:
-                return ans
+        if (self._is_special or other._is_special
```

Comment:
Needs parens to isolate the "or" part.  

    if (self._is_special or other._is_special)
        and (ans := self._check_nans(other, context=context)):